### DISCUSSION TITLE
Failed to run data generator - "Missing blockstate definitions" for every block in mod.

### DISCUSSION THREAD

**Discussion by devvoid** [FabricMC/fabric-api#2496] [Mod Dev Support]
Originally posted this as an issue, but was told it would be better suited for here.

I'm trying to setup custom model generation for some blocks I plan to add in the future. However, even registering an empty model generator causes the Data Generation task to fail. Without registering it, the mod runs perfectly fine. All of the blocks mentioned in the crash log already have blockstate and model files made in JSON.

What do I do to fix this? Asking on the Fabric discord server, someone suggested that it might be an issue with my build.gradle file, causing the data generation to not see the already-existing JSON files, but if that's the case, I have no idea how to fix it.

Crash log: https://pastebin.com/SwWrNZFm
Repository: https://github.com/devvoid/voidfurniture

**Comment by warjort** [comment]
As I said on your bug report, it is crashing because you haven't generated the blockstates/models.

The crash tells you which blocks you missed.

Try this (which makes every one of your blocks a simple cube):
````
    @Override
    public void generateBlockStateModels(BlockStateModelGenerator blockStateModelGenerator) {
    	for (Block block : BlockRegistry.BLOCKS.values()) {
    		blockStateModelGenerator.registerSimpleCubeAll(block);
    	}
    }
````
And you will probably find the problem goes away.

See BlockStateModelGenerator.register() for examples of how to this properly.

**Comment by warjort** [comment]
If you want a mixture of generated and manually written json files 
it looks you will need to override ModelProvider.run() and rewrite it to remove the validation?

**Reply by devvoid** [reply]
I'll give that a try, thank you

**Reply by Technici4n** [reply]
You can also disable strict validation if you want. Just make sure that the `fabric-api.datagen.strict-validation` property is not set.

**Comment by Technici4n** [comment]
You can remove `vmArg "-Dfabric-api.datagen.strict-validation"` from your `build.gradle` to disable strict validation.

**Reply by devvoid** [reply]
Looks like that doesn't work. Removed the line, reloaded gradle projects, fails for the same reason.
